### PR TITLE
345 - Create abstract publickey selector

### LINKED ISSUE #345: Abstract PublicKey selection
In multiple locations we have the same _public key picker_ concept replicated.
https://github.com/vyper-protocol/vyper-otc-ui/blob/cb9fc6aa53fe56a94820c64641de0704683d3d8c/src/components/molecules/ReservePicker/ReservePicker.tsx#L38-L58
https://github.com/vyper-protocol/vyper-otc-ui/blob/cb9fc6aa53fe56a94820c64641de0704683d3d8c/src/components/molecules/OraclesPicker/OraclesPicker.tsx#L30-L49
We should abstract that functionality creating a parent component that accept as inputs:
- title
- list of `{ pubkey: string, label: string }`
- `onChange` callback
- free input flag (if enabled users can provide a free publicKey)
- optionally a `renderInput` description
- eventually a category (ref #342)
The input can provide eventually a link to the explorer

### PR DESCRIPTION
**Pull request by Shikhar15606**
Fix #345

### CODE REVIEW

File: src/components/molecules/OraclesPicker/OraclesPicker.tsx [hunk 1]
```diff
@@ -111,73 +109,22 @@ const OraclePicker = ({ rateLabel: renderInputTitle, options, rateAccount, setRa
...
+				<PublicKeyPicker
+					title={renderInputTitle}
+					publicKeyLabels={options.map((o) => ({
+						label: o.title,
+						pubkey: o.pubkey,
+						category: o.category,
+						type: o.type
+					}))}
+					onChange={handleInputChange}
+					free={currentCluster === 'devnet'}
```
Comment (thegreat-vanderlinde) [inline src/components/molecules/OraclesPicker/OraclesPicker.tsx:126]: can we mantain the same mui naming here with `freeSolo`?

File: src/components/molecules/OraclesPicker/OraclesPicker.tsx [hunk 2]
```diff
@@ -111,73 +109,22 @@ const OraclePicker = ({ rateLabel: renderInputTitle, options, rateAccount, setRa
...
-					sx={{ width: 300, marginY: 2 }}
-					disableClearable
-					autoHighlight
-					selectOnFocus
-					clearOnBlur
-					handleHomeEndKeys
-					freeSolo={currentCluster === 'devnet'}
+				<PublicKeyPicker
+					title={renderInputTitle}
+					publicKeyLabels={options.map((o) => ({
```
Comment (thegreat-vanderlinde) [inline src/components/molecules/OraclesPicker/OraclesPicker.tsx:119]: can we rename the property from `publicKeyLabels` to `availableOptions` for consistency?

**Review by thegreat-vanderlinde — CHANGES_REQUESTED** [review]

File: src/components/molecules/OraclesPicker/OraclesPicker.tsx (hunk 1 above)
Comment (Shikhar15606) [reply src/components/molecules/OraclesPicker/OraclesPicker.tsx:126]: Ya sure

File: src/components/molecules/OraclesPicker/OraclesPicker.tsx (hunk 2 above)
Comment (Shikhar15606) [reply src/components/molecules/OraclesPicker/OraclesPicker.tsx:119]: that would be better

**Review by thegreat-vanderlinde — APPROVED** [review]
lgtm

**Pull request merged**